### ISSUE TITLE
no-unused-vars confused when var is re-assigned with return value of function that takes it as argument

### ISSUE THREAD

**Issue by ahocevar** [eslint/eslint#6646]
> **What version of ESLint are you using?**

3.0.1

> **What parser (default, Babel-ESLint, etc.) are you using?**

default

> **Please show your full configuration:**

https://github.com/openlayers/eslint-config-openlayers/blob/master/index.js

> **What did you do? Please include the actual source code causing the issue.**

https://github.com/openlayers/ol3/blob/9b3c4294c146dd929ecd4c8c0188dfc01580e30c/src/ol/render/canvas/canvasreplay.js#L1085-L1090

> **What did you expect to happen?**

Like in v2.x, I would expect that eslint does not complain, since `offset` is clearly used.

> **What actually happened? Please include the actual, raw output from ESLint.**

eslint complains about `offset` being defined but never used:
```
/home/travis/build/openlayers/ol3/src/ol/render/canvas/canvasreplay.js
  1085:7  error  'offset' is defined but never used  no-unused-vars
  1357:7  error  'offset' is defined but never used  no-unused-vars
```

This problem came up in openlayers/ol3#5569.

**Comment by ilyavolodin** [comment]
This should've been fixed with v3.0.1. @mysticatea could you take a quick look please?

**Comment by laurent-le-graverend** [comment]
Having the same option with the following code and version 3.0.1

```javascript
function someFunction() {
  var a = 0, i;
  for (i = 0; i < 2; i++) {
    a = myFunction(a);
  }
}
```

`'a' is defined but never used (no-unused-var)`

**Comment by mysticatea** [comment]
Thank you for this issue.
I confirmed. I will fix this.

**Comment by ahocevar** [comment]
Thanks for fixing this!
